### PR TITLE
Update to Kafka Streams 2.0 API

### PR DESCRIPTION
**Pull request by jurgispods**

### CODE REVIEW

File: lib/project/PackagingTypePlugin.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import sbt._
```
Comment (psilos) [inline lib/project/PackagingTypePlugin.scala:1]: We shouldn't need this workaround after the upgrade to Kafka Streams 2.2.0

**Comment by debasishg** [comment]
Should we target upgrading to Kafka `2.2.0` ?

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user.

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by debasishg** [comment]
@JustinPihony - I cannot find any blue mark beside the username as the linked to document specifies. Can u please help me find this problem ?

**Comment by debasishg** [comment]
@pederpansen Looks like your github profile doesn't have an email address. Hence the flagging by the validator. Could u please add an email address ?

**Comment by debasishg** [comment]
I have started integrating this PR and immediately facing issues with the latest versions ..

- the examples depend on `kafka-streams-scala` which is no longer a valid artifact. We need to use the kafka streams scala APIs from Kafka itself, which are slightly diff from the ones in `kafka-streams-scala`
- there have been changes in kafka APIs as well 
- `ZkUtils` is no longer a public class - hence we need to rewrite `kafka-local-server` using the new `AdminClient`

The effort doesn't look trivial. I am starting the work bt it will take some time to bring this to updated version of Kafka (`2.2.0`).

Thoughts ?

**Comment by jurgispods** [comment]
@debasishg I've added an email address to the commits (if this is still needed).

**Comment by debasishg** [comment]
Thanks @pederpansen ..

**Comment by lightbend-cla-validator** [comment]
Hi @pederpansen,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by jurgispods** [comment]
It took me some time to get the CLA approved, which is now done.

@debasishg Shall we get on with the issue? Do you have a branch/PR for the work you started?

**Comment by debasishg** [comment]
@pederpansen I started work on this and then got pulled away elsewhere. Check out the branch `upgrade-all` where I did some work. The `lib` now builds ok with the latest versions of all dependencies. This should be good. 

Started work on `examples`. Lots of stuff to work on here. I have changed some of the versions - will continue to work here. 

One thing that I plan to do is to replace `local-kafka-server` with https://github.com/embeddedkafka/embedded-kafka ..

It will be great if u can help on any of these fronts ..

**Comment by davideicardi** [comment]
I can also try to help on this task.
Can you provide some advice on where I can help?

If I understand correctly we should work on the `upgrade-all` branch? We should convert examples and replace the local-kafka-server?

thanks

**Comment by debasishg** [comment]
@davideicardi You are correct. Unfortunately I have not been able to work on this because of serious lack of time. It would be great if u can take a look.

**Comment by davideicardi** [comment]
I think that we should also use `kafka-streams-scala` library. There is any reason to not use it?
It is probably a lot of work, but it should make sense...

**Comment by debasishg** [comment]
`kafka-streams-scala` has been merged to official `kafka` project (https://github.com/apache/kafka/tree/trunk/streams/streams-scala).

**Comment by ennru** [comment]
We will not upgrade the library for later versions.

**Pull request closed without merging**